### ISSUE TITLE
Nonsquare MO coefficient matrices break AFQMC converter 

### ISSUE THREAD

**Issue by shivupa** [QMCPACK/qmcpack#2291]
**Describe the bug**
If MOs were dropped due to prevent linear dependency, the MO coefficient matrix will be of shape `(num_ao, num_ao-num_dropped)`. The AFQMC converter doesn't handle this case.

**To Reproduce**
1. release version or git commit hash being built
latest develop 02078f8eabd9c42c10d772e35e1b2c4bae09a599

2. cmake command
Doesn't require compilation

3. full program/test invocation command
`python ~/Documents/qmcpack/utils/afqmctools/bin/pyscf_to_afqmc.py -i h2o.chk -o qmc_h2o4.structure.xml -w qmc_h2o4.in.xml -q qmc_h2o4.qmc.in.xml --ao`
or 
`python ~/Documents/qmcpack/utils/afqmctools/bin/pyscf_to_afqmc.py -i h2o.chk -o qmc_h2o4.structure.xml -w qmc_h2o4.in.xml -q qmc_h2o4.qmc.in.xml `
4. additional steps
This calculation was using a [molden file](https://gist.github.com/shivupa/447ebe083d245374bb6daf7c7f346751) created from a psi4 calculation.
To create a pyscf chkfile from a molden:

```
import pyscf.tools as pt
from pyscf.scf.chkfile import dump_scf

mol, mo_energy, mo_coeff, mo_occ, irrep_labels, spins = pt.molden.load("anion_allelec_res_4.0_no_sdci_psi4_norev.molden")
print(mo_coeff.shape)
dump_scf(mol, 'h2o.chk', 0.0, mo_energy, mo_coeff, mo_occ)
```

**Expected behavior**
The pyscf chkfile would be converted for use with qmcpack.

**System:**
 - system name [e.g. fusiont5, summit]
 - modules loaded [e.g. output of `module list`]
 - other systems where this is reproducible [e.g. "my laptop", "none"]
my workstation(Python 3.8.1), university cluster(Python 3.7.3 ).

**Additional context**
Discussed with @fdmalone, and I'm opening an issue since the bug was confirmed.

**Comment by fdmalone** [comment]
This was an oversight on my side. It should be a quick fix.

**Comment by fdmalone** [comment]
@shivupa  I opened a PR to fix this.

A couple of points:
1. -o and -w are hdf5 files for the hamiltonian and wavefunction respectively, not xml files. The help info for -w has a typo which I will fix. -w isn't necessary unless you want to write the wavefunction to a different file than the hamiltonian.
2. It's a bit misleading and I will change the variable name (the help message is clearer), but --ao will transform the integrals and wavefunction to the orthogonalised ao basis, not the typical non-orthogonal ao basis. This option is required if you have a UHF-like wavefunction. By default the script will use the matrix in the mo_coeff field of the scf.chk file to rotate the integrals to this basis, and will set the trial Slater Matrix to be an (NMO,N) dimensional matrix with 1s on the diagonal.

**Comment by shivupa** [comment]
Thanks for the rapid response! The first point makes sense. Just a quick question about the second point, for the orthogonal AO case the matrix used is the X matrix right? Is this the S^{-1/2} matrix from symmetric orthogonalization? 
 ```
if ortho_ao:
        X = scf_data['X']
else:
    if scf_data['isUHF']:
         print(" # UHF integrals are not allowed. Use ortho AO option (-a/--ao).")
         sys.exit()
    X = scf_data['mo_coeff']
```

**Comment by shivupa** [comment]
Ok I think I understand I should be looking at [this](https://github.com/QMCPACK/qmcpack/blob/93cb4243d551ad18216634b2d654f6fa22606aea/utils/afqmctools/afqmctools/wavefunction/mol.py#L52)

**Comment by shivupa** [comment]
Fixed by #2292

**Comment by fdmalone** [comment]
Note that if you want to use the oao basis you will need to write the transformation matrix to the scf.chk file as follows:

```python
from afqmctools.utils.linalg import get_ortho_ao_mol
X = get_ortho_ao_mol(S, LINDEP_CUTOFF=cutoff)
with h5py.File('scf.chk', 'r+') as fh5:
    fh5['scf/orthoAORot'] = X
```
where cutoff is the linear dependency threshold used in the SCF calculation. I need to submit a PR to deal with this rectangular matrix too.
